[DRAFT] feat: security hub integration in terraform provider

hyadav1396 opened the pull request:
# Description

Add support for Aws Security Hub integration.
Go client side PR : https://github.com/newrelic/newrelic-client-go/pull/1300

Jira ticket : https://new-relic.atlassian.net/browse/NR-412555

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

Note : Not Updating the documentation as of now in this PR. Documentation changes requires review from the content team.
So I'll raise another PR on the documentation change when this new feature goes live.

## Checklist:

- [x] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] My code is formatted to [Go standards](https://go.dev/blog/gofmt)
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes. Go [here](https://github.com/newrelic/terraform-provider-newrelic/blob/main/CONTRIBUTING.md#testing) for instructions on running tests locally.

## How to test this change?

Please describe how to test your changes. Include any relevant steps in the UI, HCL file(s), commands, etc

- Performed the testing for creating/updating Aws Security Hub integration locally for an NR account.
- Updated the go.mod file to use local go-client.
- updated the changed to newrelic.tf file to test :
`resource "newrelic_cloud_aws_link_account" "foo" { 
    arn = "arn:aws:iam::22263*******:role/NewRelicInfrastructure-Integrations"
     metric_collection_mode = "PULL" 
     name = "22263******" 
  }`
  
  and 
  
  `resource "newrelic_cloud_aws_integrations" "bar" {
  linked_account_id = newrelic_cloud_aws_link_account.foo.id
  security_hub{
    metrics_polling_interval = 43200
    aws_regions              = ["us-east-1"]
  }`
  

- Was able to successfully create/update/delete the security hub integration.
- Attaching some screenshot reference for the same :
![image](https://github.com/user-attachments/assets/ea1a935e-90b0-4bdf-bde0-7f6f8991d498)

Aashirwadjain commented:
Hey @hyadav1396, Can you please update the update the documentation to include this new integration support in `Supported AWS Integrations` section?


https://github.com/newrelic/terraform-provider-newrelic/blob/main/website/docs/r/cloud_aws_integrations.html.markdown?plain=1#L100

vinay-newrelic committed:
docs: enhances newrelic_api_access_key resource documentation (#2884)

vinay-newrelic committed:
fix(synthetics): disable count check to track changes outside of terraform scope in synthetics_secure_credential (#2891)

shashank-reddy-nr committed:
fix: resolve distributed tracing flag misfunction bug (#2889)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.62.1

vinay-newrelic committed:
chore: resolved deprecated lint checks github action (#2893)

amandeepmalik committed:
feat: enable workflow automation entities creation

amandeepmalik committed:
feat: updated html markdown

vinay-newrelic committed:
feat(cloud): added AWS AutoDiscovery Slug for configure-integration API (#2898)

Co-authored-by: pchawla-NR <pchawla@newrelic.com>

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.63.0

Aashirwadjain committed:
feat(tooltip-support): add tooltip support in dashboard rawConfiguration (#2911)

Aashirwadjain committed:
fix(entity-tags): resolve `newrelic_entity_tags` resource issue (#2913)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.64.0

pranav-new-relic committed:
feat(dashboards): add support for multi account queries in widgets (#2914)

Co-authored-by: vgodhala <vgodhala@newrelic.com>

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.65.0

pchawla-NR committed:
feat(cloud): corrected Documentation for Auto-Discovery (#2917)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.66.0

pranav-new-relic committed:
feat(pipelinecontrol): add a new resource to manage Pipeline Cloud Rules (#2927)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.67.0

pranav-new-relic committed:
feat(nrqldroprule): fully deprecate resource, add field to map to PCRs, update EOL guides (#2929)

feat(nrqldroprule): fully deprecate resource, add field to map to PCRs, update EOL guides (#2929)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.68.0

pranav-new-relic committed:
feat(apiaccess): revamp and change API Key visibility, newrelic_api_access_keys (#2931)

feat(apiaccess): revamp and change API Key visibility, newrelic_api_access_keys (#2931)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.69.0

nr-mjuvvadi committed:
docs(synthetics): update examples with `depends_on` for race condition with monitor and secure credential (#2935)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.69.1

msivant committed:
feat(cloud): add `newrelic_cloud_oci_link account` resource, associated onboarding modules (#2936)

feat(cloud): add `newrelic_cloud_oci_link account` resource, associated onboarding modules (#2936)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.70.0

hrai-nr committed:
fix(cloud): make subnet optional in oci module (#2938)

Co-authored-by: pranav-new-relic <pnandula@newrelic.com>

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.70.1

Aashirwadjain committed:
fix(newrelic_one_dashboard): add plan-time validation for `account_ids` in `variable{}` block (#2930)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.70.2

Aashirwadjain committed:
fix(dashboard): add billboard settings support in newrelic_one_dashboard resource (#2916)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.70.3

pranav-new-relic committed:
fix(deps): upgrade v2 to v3 to allow go.dev listing (#2941)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.70.4

msivant committed:
fix(oci): update Docker image path and newrelic_endpoint validation (#2942)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.70.5

sarvetiNr committed:
fix(aws): add a variable to the AWS module to help customers opt in or out of creation of a configuration recorder resource (#2944)

fix(aws): add a variable to the AWS module to help customers opt in or out of creation of a configuration recorder resource (#2944)

rkumarregar committed:
fix(aws): remove aws_auto_discovery from the example module as it is a separate integration (#2945)

fix(aws): remove aws_auto_discovery from the example module as it is a separate integration (#2945)

sarvetiNr committed:
fix(output-format-example): otel version upgrade changes (#2946)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.70.6

wulczer committed:
feat(alerts): add support for target entity in NRQL conditions (#2950)

feat(alerts): add support for target entity in NRQL conditions (#2950)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.71.0

pranav-new-relic committed:
feat(alerts): revert add support for target entity in NRQL conditions #2950 (#2951)

feat(alerts): revert add support for target entity in NRQL conditions #2950 (#2951)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.72.0

pranav-new-relic committed:
fix(nrqldroprules): fix entity mapping with pipeline cloud rules, allow accurate retry mechanism (#2955)

fix(nrqldroprules): fix entity mapping with pipeline cloud rules, allow accurate retry mechanism (#2955)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.72.1

pranav-new-relic committed:
fix(nrqldroprules): add ci migration guide phase 1 via terraform script (#2957)

fix(nrqldroprules): add ci migration guide phase 1 via terraform script (#2957)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.72.2

pranav-new-relic committed:
fix(nrqldroprules): brief corrections to the migration guide (#2958)

fix(nrqldroprules): brief corrections to the migration guide (#2958)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.72.3

pranav-new-relic committed:
feat(nrqldroprules): rebuild the guide explaining the Drop Rules EOL, remediations, automated helper approaches (#2960)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.73.0

wulczer committed:
feat(alerts): add support for target entity in NRQL conditions (#2959)

feat(alerts): add support for target entity in NRQL conditions (#2959)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.74.0

adarshnewrelic committed:
feat(azure): introduce azure autodiscovery integration for limited preview (#2943)

feat(azure): introduce azure autodiscovery integration for limited preview (#2943)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.75.0

Aashirwadjain committed:
fix(dashboards): add support for `showApplyAction` in variable options (#2964)

fix(dashboards): add support for `showApplyAction` in variable options (#2964)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.75.1

shubhamgupta-dev committed:
fix(oci): remove the usage of svcUsername field (#2966)

fix(oci): remove the usage of svcUsername field (#2966)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.75.2

pranav-new-relic committed:
fix(deps): update Go Client to v2.73.2 to support OCI Schema Change (#2969)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.75.3

msivant committed:
fix(oci): store Region and Terraform Details Post Metrics and Logs Instrumentation (#2961)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.75.4

tomtupy-nr committed:
feat(alerts): add support for outlier nrql conditions (#2977)

Co-authored-by: pranav-new-relic <pnandula@newrelic.com>

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.76.0

pranav-new-relic committed:
fix(accountmanagement): revise working approach to manage accounts, improve read efficiency (#2974)

pranav-new-relic committed:
chore(deps): upgrade Go Client to v2.74.1 (#2980)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.76.1

msivant committed:
fix(oci): implement terraform module for WIF onboarding (#2971)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.76.2

pranav-new-relic committed:
fix(accountmanagement): curb organization query induced bug (#2986)

fix(accountmanagement): curb organization query induced bug (#2986)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.76.3

pranav-new-relic committed:
fix(accountmanagement): fix outliers in the resource, allow edge case with import, etc (#2991)

fix(accountmanagement): fix outliers in the resource, allow edge case with import, etc (#2991)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.76.4

pranav-new-relic committed:
fix(accountmanagement): minor bug fix to the active-canceled account read mechanism (#2997)

fix(accountmanagement): minor bug fix to the active-canceled account read mechanism (#2997)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.76.5

pranav-new-relic committed:
fix(nrqldroprules): update EOL date, in light of the new effective date, June 30, 2026 (#3001)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.76.6

pranav-new-relic committed:
fix(deps): bump up versions of key vulnerabilities, New Relic Go Client (#3003)

fix(deps): bump up versions of key vulnerabilities, New Relic Go Client (#3003)

nr-developer-toolkit committed:
chore(changelog): update CHANGELOG for v3.76.7

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/newrelic/terraform-provider-newrelic?pullRequest=2888) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/newrelic/terraform-provider-newrelic?pullRequest=2888) before we can accept your contribution.<br/>**3** out of **16** committers have signed the CLA.<br/><br/>:white_check_mark: Aashirwadjain<br/>:white_check_mark: sarvetiNr<br/>:white_check_mark: pranav-new-relic<br/>:x: shashank-reddy-nr<br/>:x: vinay-newrelic<br/>:x: amandeepmalik<br/>:x: nr-developer-toolkit<br/>:x: pchawla-NR<br/>:x: nr-mjuvvadi<br/>:x: hrai-nr<br/>:x: msivant<br/>:x: rkumarregar<br/>:x: adarshnewrelic<br/>:x: tomtupy-nr<br/>:x: wulczer<br/>:x: shubhamgupta-dev<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/newrelic/terraform-provider-newrelic?pullRequest=2888) it.</sub>

pranav-new-relic committed:
fix(deps): fix vulnerabilities, January 2026 edition (#3002)

pranav-new-relic committed:
chore: squashed commits

pranav-new-relic commented:
looks like this PR cannot be recovered; closing this, to be replaced with #3008
